src/language/dafnyLanguageClient.ts

```diff
@@ -51,6 +52,20 @@ function getMarkGhostStatementsArgument(): string {
   return `--ghost:markStatements=${Configuration.get<string>(ConfigurationConstants.LanguageServer.MarkGhostStatements)}`;
 }
 
+function getDafnyPluginsArgument(): string[] {
+  const plugins = Configuration.get<string[]>(ConfigurationConstants.LanguageServer.DafnyPlugins);
+  if(plugins === null || !Array.isArray(plugins)) {
+    return [];
+  }
+  const result = [];
+  for(const i in plugins) {
```

Comment:
It should generally be avoided to use `for(... in ...)` to iterate over the indices of an array. For example, the order of the keys (which are the indices for arrays) is not consistent:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#array_iteration_and_for...in

Threfore, prefer using a conventional for loop to iterate over the indices.

However, I think this whole for-loop can be neatly refactored into a map and filter call. Something like this:
```js
return plugins
  .filter(plugin => plugin != null && plugin !== '')
  .map((plugin, index) => `--dafny:plugins:${index}=${plugin}`);
```

Reply:
Well, I prefer comprehensions too. I also added parentheses around your expression because the semicolon is optional in JavaScript.

Reply:
> Well, I prefer comprehensions too. I also added parentheses around your expression because the semicolon is optional in JavaScript.

Because the semicolons are optional in JavaScript/TypeScript, I configured eslint to enforce them. It's way to easy to forget about some implicit semicolon, especially when context switching between multiple languages.